File: app/src/main/java/com/chooloo/www/callmanager/entity/RecentCall.java
package com.chooloo.www.callmanager.entity;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.util.Date;

import static android.provider.CallLog.Calls.INCOMING_TYPE;
import static android.provider.CallLog.Calls.MISSED_TYPE;
import static android.provider.CallLog.Calls.OUTGOING_TYPE;
import static android.provider.CallLog.Calls.REJECTED_TYPE;
import static android.provider.CallLog.Calls.VOICEMAIL_TYPE;

public class RecentCall {

    private final long callId;
    private final int callType;
    private final Date callDate;
    private final String number;
    private final String callDuration;

    public static final int TYPE_OUTGOING = OUTGOING_TYPE;
    public static final int TYPE_INCOMING = INCOMING_TYPE;
    public static final int TYPE_MISSED = MISSED_TYPE;
    public static final int TYPE_VOICEMAIL = VOICEMAIL_TYPE;
    public static final int TYPE_REJECTED = REJECTED_TYPE;

    @Retention(RetentionPolicy.SOURCE)
    @IntDef({TYPE_OUTGOING, TYPE_INCOMING, TYPE_MISSED, TYPE_VOICEMAIL, TYPE_REJECTED})
    public @interface CallType {
    }

    public RecentCall(long id, String number, String duration, Date date, int type) {
        this.callId = id;
        this.number = number;
        this.callDuration = duration;
        this.callDate = date;
        this.callType = type;
    }

    public long getCallId() {
        return this.callId;
    }

    public String getCallerNumber() {
        return this.number;
    }

    public int getCallType() {
        return this.callType;
    }

    public String getCallDuration() {
        return this.callDuration;
    }

    public Date getCallDate() {
        return this.callDate;
    }
}

File: app/src/main/java/com/chooloo/www/callmanager/entity/RecentCall.kt
package com.chooloo.www.callmanager.entity

import android.provider.CallLog
import androidx.annotation.IntDef
import java.util.*

class RecentCall(
        val callId: Long? = null,
        val callerNumber: String? = null,
        val callDuration: String? = null,
        val callDate: Date? = null,
        val callType: Int? = null
) {

    @kotlin.annotation.Retention(AnnotationRetention.SOURCE)
    @IntDef(TYPE_OUTGOING, TYPE_INCOMING, TYPE_MISSED, TYPE_VOICEMAIL, TYPE_REJECTED)
    annotation class CallType

    companion object {
        const val TYPE_OUTGOING = CallLog.Calls.OUTGOING_TYPE
        const val TYPE_INCOMING = CallLog.Calls.INCOMING_TYPE
        const val TYPE_MISSED = CallLog.Calls.MISSED_TYPE
        const val TYPE_VOICEMAIL = CallLog.Calls.VOICEMAIL_TYPE
        const val TYPE_REJECTED = CallLog.Calls.REJECTED_TYPE

        val UNKNOWN = RecentCall(callerNumber = "Unknown")
    }

}